bootstrap.ps1

```diff
@@ -7,13 +7,28 @@ Push-Location (Join-Path $PSScriptRoot "build")
     .\prerequisites.ps1
 Pop-Location
 
+Push-Location (Join-Path $PSScriptRoot "./src/Simulation/qdk_sim_rs")
+    # We use dotnet-script to inject the version number into Cargo.toml,
+    # so we go on ahead here and restore any missing tools.
+    # Since that Cargo.toml is referenced by CMake lists in the QIR
+    # runtime, this injection has to be the first thing we do.
+    dotnet tool restore
+    dotnet script inject-version.csx -- `
+        --template Cargo.toml.template `
+        --out-path Cargo.toml `
+        --version $Env:NUGET_VERSION;
+Pop-Location
+
 if (-not (Test-Path Env:AGENT_OS)) {
     if ($Env:ENABLE_NATIVE -ne "false") {
         Write-Host "Build release flavor of the native simulator"
         $Env:BUILD_CONFIGURATION = "Release"
         Push-Location (Join-Path $PSScriptRoot "src/Simulation/Native")
             .\build-native-simulator.ps1
         Pop-Location
+        Push-Location (Join-Path $PSScriptRoot "src/Simulation/qdk_sim_rs")
+            .\build-qdk-sim-rs.ps1
```

Comment:
Since we don't have logic in build/prerequisites.ps1 that will install Rust automatically, we should either include logic in build-qdk-sim-rs.ps1 to no-op if cargo is not in the path or update the top-level readme.md to indicate that manually installing Rust is a prerequisite. As it is, it looks like this will unconditionally run and then fail during bootstrap for users without Rust installed.

Reply:
Sounds good, let me add that, then. If it's alright, I'll similarly gate that on non-CI usage, so that CI builds fail correctly if prerequisites are missing?